core/src/main/java/org/infinispan/transaction/tm/DummyTransaction.java

```diff
@@ -16,34 +16,71 @@
 import javax.transaction.xa.Xid;
 import java.util.ArrayList;
 import java.util.Collection;
-import java.util.HashSet;
+import java.util.Collections;
 import java.util.List;
-import java.util.Set;
+
+import static java.lang.String.format;
 
 /**
  * @author bela
  * @since 4.0
  */
 public class DummyTransaction implements Transaction {
+   /*
+    * Developer notes:
+    *
+    * => prepare() XA_RDONLY: the resource is finished and we shouldn't invoke the commit() or rollback().
+    * => prepare() XA_RB*: the resource is rolled back and we shouldn't invoke the rollback().
+    * //note// for both cases above, the commit() or rollback() will return XA_NOTA that we will ignore.
+    *
+    * => 1PC optimization: if we have a single XaResource, we can skip the prepare() and invoke commit(1PC).
+    * => Last Resource Commit optimization: if all the resources (except last) vote XA_OK or XA_RDONLY, we can skip the
+    * prepare() on the last resource and invoke commit(1PC).
+    * //note// both optimization not supported since we split the commit in 2 phases for debugging
```

Comment:
Also, Infinispan decides whether to do 1PC based on its own config and pretty much ignores the TM's decision :)